### ISSUE TITLE
Make adding exceptions and editing filter lists easier for common user

### ISSUE THREAD

**Issue by SteveH66** [gorhill/uBlock#1323]
Using Firefox 44.0
example site: news.cnn.com
problem :  With uBlock Origin blocking the site, the comments are disabled.  After hours wasted on this problem I found out the following : 1. The commenting platform is Livefyre.  2. There are **many** elements from Livefyre to this site, and the 'common user' would not be able to figure out which ones were necessary to allow in order to be able to read/post comments on a site.  4. For some reason, some filter list writers/maintainers/whatever consider the comment section to be a nuisance on (at least some) sites - I have found that it is hit and miss whether comments will show or not show on a site without turning off add blocking on the site.  5. The wiki for your project is very thin on instructions for creating exception filters, actually the only way I found out how to do an exception filter was to go to the AdBlock Plus page you linked to for their compatible filter syntax or whatever.

The 'average user' isn't going to want to run to the Adblock Plus forums every time they go to a site and can't see the comments section (or something else they want to see on a site that a filter list writer has arbitrarily determined needs to be blocked) and ask someone there to help them create an exception filter.  I would suspect, and correct me if I am wrong here, that if the people who answer questions in the Ablock Plus forums know you are using uBlock for your ad blocking instead of their extension, they might get tired of answering those questions fairly quickly as well.  uBlock Origin has no user support forum to ask these questions in though, posting here in issues was the only solution I could think of.

What elements need to be unblocked in order to make something like the comments section work?  xhr? Script? In-line script? Images? Frames? CSS?  Some of them or all of them?  If only some, which ones?  I used the element picker to select everything for the frame for the comments section.  I created a rule, the only rule it seems to allow creation of is blocking rules, not exception rules.  I modified this rule with @@ to make it an exception rule.  Didn't work.  Then, I went in and messed around with dynamic blocking, clicking all of the elements (quite a long list) to green, which if I understand your documentation means allowed.  Didn't work.

There needs to be an easy method incorporated into the extension so that a 'common user' with no programming or web page editing skills/knowledge can make an exception so that they can show something they want to see, whether a list writer/maintainer thinks it is an annoyance or not.  I don't know enough about programming or server side programming (whatever you would call it) to know what this would be.  Maybe a way to use the Element Picker to select an area and make an exception rule to allow it?  Maybe an easy way to look up what lists have that particular function blocked, and then easily write an exception rule that overrides the blocking rule?  I don't know.  Even though I am sure the author/programmer of this extension is a busy person, maybe he could get some help from knowledgeable people here to write more extensive documentation?  Maybe open up a support forum somewhere like Adblock Plus has, and the knowledgeable people here take the time to scan the forums from time to time?

I really do like your extension, and I know there are thousands of others out there who do too.  But this is a very frustrating problem, and one might want to be able to discover how to handle some of these things themselves instead of having to go to the Adblock Plus forum to ask for help.  Anyway, I hope that this post will be taken as constructive, it isn't meant to denigrate your extension, which as I said I think is excellent.  And I would appreciate you taking a look into this issue.  Thank you

**Comment by SteveH66** [comment]
OK, so I got some help on the problem over in the ABP forum.  Found out a few things.  One thing I found out is, as I suspected, they do not care much to help uBlock users with support questions, mapx did help me find the problem, even explained how he did it in ABP, and told me to take any support questions for other products to their site or forums.

He told me that he found the culprit for the problem by using the ABP 'show list of blockable items' option in ABP, disabled each blocking rule until he found the guilty party, and then made an exception rule for me that overrode the blocking rule.  Does uBlock have something similar to 'show list of blockable items'?  If it doesn't, it sounds like it would be HIGHLY advisable to add it.  Once explained, that technique sounded like something that any user who is fairly tech proficient might be able to use to track this kind of thing down.

And you would need to be very proficient, to find it without something like 'show list of blockable items'.  It turned out that the problem was indeed a list based problem.  For some reason, Fanboy decided that downloads.mailchimp.com is an annoyance and he put it on his Annoyance List.  Even if I were very proficient at coding (programs & web based source), with the tools uBlock has that I know of I would have never thought to look for downloads.mailchimp.com as the source of the problem for why you couldn't get comments from Livefyre, which is at the domains www.livefyre.com and www.livefyre.co - a web coder might know right away to look at mailchimp.com as the guilty party for why you're not getting comments from livefyre.com but I know I would have never guessed it, and my guess is your average Joe web-coder wouldn't have even guessed that one.

So anyway, there is an example of some tools you might think about adding, or if you have equivalent tools in uBlock some more documentation in the Wiki about them (how to find them, what they are, how to use them) might come in pretty handy.  And I did find your support forum, so hopefully people who are good at this kind of thing and really know your extension will spend a lot of time there helping people with questions.  It just seems that ABP is a bit more user friendly and intuitive for the common user than your extension, unless a person can do the same things with your extension as they can with ABP but maybe in a different way, and you can get that information out there to the new users.  Maybe it's just that ABP has been around longer, so people using ad blocking are just used to doing things in that way.  As I said in my OP, I'm not trying to nit pick or run down your extension, I have liked it quite a bit since I switched over to it to try it out, and I know many others like it as well as it gets some good reviews in articles and user reviews.  I'm just hoping the documentation can get better, and the support forum gets well known, and lots of experienced uBlock 'pros' will be able to help out over there to guide us 'new fish' :)  ABP has had a lot of time to get their forums established and get helpful people answering questions over there, so I hope you can catch up to them on that quickly.  Thanks

**Comment by gorhill** [comment]
Closing and marking as _invalid_, as per [CONTRIBUTING](https://github.com/gorhill/uBlock/blob/master/CONTRIBUTING.md).

**Comment by SteveH66** [comment]
Ok, this is not a bug per se, so I am going to do a post over in your support forum.  I will give you all the information you ask in your CONTRIBUTE page, except for the things that your wiki here nor anywhere else I have been able to find on the internet tell you HOW to find out through your extension.

**Comment by gorhill** [comment]
Just to be clear, I won't clone ABP's way of disabling filters -- disabling a filter in ABP causes the filter to be disabled _everywhere_.

The way to deal with bad filters in uBO is to _override_ them, and this can be done on a per-site basis. Bottom-line is if you absolutely want to be able to disable filters like in ABP, just use ABP, uBO has a different approach.

If you provide a real, concrete case, I would show you how to deal with this in uBO, but you did not provide such case.

**Comment by gorhill** [comment]
Mini tutorial using your specific case: https://discourse.mozilla-community.org/t/support-ublock-origin/6746/2?u=gorhill

**Comment by SteveH66** [comment]
Well gorhill, I wouldn't expect you to clone that particular behavior, as I agree with your assessment that it might not be a good idea to disable a filter globally for all sites instead of on a per-site basis as you said.  In many cases I can see where your approach sounds to be a better option if something is causing problems like allowing ads on a site or something like that, but on another site it isn't causing problems and not having it working breaks site functionality.  So I like your approach for that better.  Thank you for taking the time to do this mini tutorial, and I will definitely check it out and learn how to do per-site overriding of a filter.

**Comment by gorhill** [comment]
I will post another tutorial for dealing with unbreaking page using another approach (logger) soon, hopefully before next week.

**Comment by SteveH66** [comment]
gorhill,

Thanks for that very useful tutorial you made, that should help myself and a lot of other novices.  I look forward to your tutorial for how to do it by using logger.  I may have stumbled across your method for using logger the other night, not even knowing what I was doing, when I was trying to enable comments on that page.  I didn't know it was dynamic rules, though.  And in messing around with that, I may have done some dynamic rules that I shouldn't have, because the page allowed some things it wasn't originally allowing that a user on a metered bandwidth plan might not want to happen - since they are paying by the GB or whatever on their data plan.

The video at the top (and on the right side when you scroll down the page) started to play automatically (it wasn't auto-playing at first), and that Twitter image about the percentage of posts for each candidate started loading when it wasn't loading before.  I don't know if your method of dynamic rules in uOB would allow this behavior, if that allow rule I got from ABP forum ( @@/downloads.mailchimp.com/js/mc-validate.js$domain=money.cnn.com ) did it, or if I did it messing around with dynamic rules not knowing what I was doing.  I will post about it over on the tutorial you wrote.

Would your logger method be to click on the green block (in the 4th column of logger I believe) to allow each site, one at a time, and then when you find what is blocking comments you go to the Dashboard and on the My rules tab you click on the save button under Temporary rules?
